crates/apps/src/lib/node/ledger/storage/rocksdb.rs

```diff
@@ -231,14 +235,38 @@ impl RocksDB {
 
         if let Some(new_value) = new_value {
             let new_val_key = key_prefix
-                .push(&"new".to_owned())
+                .push(&NEW_DIFF_PREFIX.to_owned())
                 .map_err(Error::KeyError)?
                 .join(key)
                 .to_string();
             self.0
                 .put_cf(cf, new_val_key, new_value)
                 .map_err(|e| Error::DBError(e.into_string()))?;
         }
+
+        // If not persisting the diffs, remove the existing diffs from two block
+        // heights earlier than `height`
+        if !persist_diffs {
```

Comment:
I don't think this works. This deletes a key with height `height - 2`. But say that we write a non-diff key at height 5 and than the next write happens at height 7, than we'll try to delete the keys `6/old/` and `6/new` which don't exist. Instead we should be deleting the keys `5/old` and `5/new`. This can build up and we can end up with quite a few stale keys in storage